packages/adblocker/src/lists.ts

```diff
@@ -189,18 +203,45 @@ export function parseFilters(
       const filter = NetworkFilter.parse(line, config.debug);
       if (filter !== null) {
         networkFilters.push(filter);
+        if (preprocessor !== null) {
+          preprocessors.set(filter.getId(), preprocessor);
+        }
       }
     } else if (filterType === FilterType.COSMETIC && config.loadCosmeticFilters === true) {
       const filter = CosmeticFilter.parse(line, config.debug);
-      if (filter !== null) {
-        if (config.loadGenericCosmeticsFilters === true || filter.isGenericHide() === false) {
-          cosmeticFilters.push(filter);
+      if (
+        filter !== null &&
+        (config.loadGenericCosmeticsFilters === true || filter.isGenericHide() === false)
+      ) {
+        cosmeticFilters.push(filter);
+        if (preprocessor !== null) {
+          preprocessors.set(filter.getId(), preprocessor);
```

Comment:
Shouldn't we register filters in preprocessors only if we are in a preprocessor block? Unless I misunderstood what this does (which is likely), my understanding is that the vast majority of filters should have nothing to do with preprocessors, and as a result we should not have any overhead for these.

While parsing the lists, whenever we find a line that is a preprocessor block we should then start registering the filters, until an `!#else` (this means we need to create a new processor for this other condition negated or handle it in the first preprocessor) or `!#endif` (where we "close" the current preprocessor).

Reply:
I think the current logic is handled as intended. The type of local variable `preprocessor` is `Preprocessor | null`, so we can check if we're in preprocessor block by checking `null`.

In the extend to support nested preprocessor blocks like:

```adb
!#if a

!#if b
!#endif

!#endif
```

I think we can change the type into `Preprocessor[]` and check the length instead.